### PR TITLE
chore/SC-116732/add-Orson-platform

### PR DESCRIPTION
**Pull request by keeramis**
## Description

- Adds the new Orson platform `muon`
- Fix P2 display name as per the latest changes in the device-constants library

## How to Test
 
Prepare an Orson device with a device-os version that supports inspecting via serial (supports the 's' command via serial).
Run `npm start -- serial inspect`

## Related Issues / Discussions

https://app.shortcut.com/particle/story/116732/support-orson-platform-in-cli-tools

## Completeness

- [x] User is totes amazing for contributing!
- [x] Contributor has signed [CLA](https://docs.google.com/a/particle.io/forms/d/1_2P-vRKGUFg5bmpcKLHO_qNZWGi5HKYnfrrkd-sbZoA/viewform)
- [x] Problem and solution clearly stated
- [x] Tests have been provided
- [x] ~Docs have been updated~ Docs will be updated during release
- [x] CI is passing

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -52,7 +52,7 @@
     }
   ],
   "dependencies": {
-    "@particle/device-constants": "^2.0.0",
+    "@particle/device-constants": "^3.1.6",
     "binary-version-reader": "^1.1.1",
```
Comment (monkbroc) [inline package.json:56]: You'll need to bump binary-version-reader and particle-usb to keep the versions consistent

**Comment by monkbroc** [comment]
While you're in there, can you add an alias `photon2: PlatformId.P2` to `src/cmd/cloud.js` please? That will allow people to do `particle compile photon2`

File: package.json (hunk 1 above)
Comment (keeramis) [reply package.json:56]: thanks @monkbroc! A couple of thoughts from my side:
1. I will bump up the node/npm versions for the particle-cli. The `binary-version-reader : 1.1.1` still worked with node v12 and npm v6. The more latest versions of BVR need [upgraded versions of node and npm](https://github.com/particle-iot/binary-version-reader/commit/0aa18f2181da9ddf4b4ebaa102a216b6d418f1e9) 

2. Is this the correct order? Add necessary changes to BVR and Particle-USB that supports Orson and then continue with this PR.

**Review by hugomontero — APPROVED** [review]
👍 
I ran the follow tests:
1. Inspect a `p2` by running `npm start -- serial inspect`: The display name is correct.
2. compile for `p2`, `muon`, `photon2` by running `npm start -- compile  `platform`: No errors regarding unsupported platform.

**Review by scott-brust — APPROVED** [review]
See comments [here](https://app.shortcut.com/particle/story/116732/support-orson-platform-in-cli-tools#activity-116793)

**Pull request merged**